src/python/dbsClient/apis/dbsClient.py

```diff
@@ -10,6 +10,157 @@
+    """
+    rows = []
+    for row in results:
+        if attr in row and not isinstance(row[attr], list):
+            rows.append(row[attr])
+    if len(rows) > 0:
+        return [{attr: rows}]
+    return results
+
+def aggRuns(results):
+    """
+    performs runs API aggregation
+
+    :param: results: list JSON records
+    :type: list
+    :return: aggregated list of JSON records
+    """
+    return aggAttribute(results, 'run_num')
+
+def aggReleaseVersions(results):
+    """
+    performs release version API aggregation
+
+    :param: results: list JSON records
+    :type: list
+    :return: aggregated list of JSON records
+    """
+    return aggAttribute(results, 'release_version')
+
+def aggDatasetAccessTypes(results):
+    """
+    performs dataset access types API aggregation
+
+    :param: results: list JSON records
+    :type: list
+    :return: aggregated list of JSON records
+    """
+    return aggAttribute(results, 'dataset_access_type')
+
+def aggFileLumis(results):
+    """
+    performs filelumis API aggregation
+
+    :param: results: list JSON records
+    :type: list
+    :return: aggregated list of JSON records
+    """
+    output = []
+    file_run_lumi = {}
+    event_ct = False
+    if results and 'event_count' in results[0]:
+        event_ct = True
+    for rec in results:
+        run = rec['run_num']
+        lfn = rec['logical_file_name']
+        lumi = rec['lumi_section_num']
+        if isinstance(lumi, list):
+            break
+        if event_ct:
+            file_run_lumi.setdefault((lfn, run), []).append([rec['lumi_section_num'], rec['event_count']])
+        else:
+            file_run_lumi.setdefault((lfn, run), []).append(rec['lumi_section_num'])
+    # if we get results from Python server no aggregation is required
+    if len(file_run_lumi) == 0:
+        return results
+    # if we got results from GO server we need to perform aggregation of results based on file/run pair
+    for key in file_run_lumi.keys():
+        val = file_run_lumi[key]
+        if event_ct:
+            lumi=[]
+            event=[]
+            for item in val:
+                lumi.append(item[0])
+                event.append(item[1])
+            rec = {'logical_file_name':key[0], 'run_num':key[1], 'lumi_section_num':lumi, 'event_count':event}
+        else:
+            rec = {'logical_file_name':key[0], 'run_num':key[1], 'lumi_section_num':val}
+        output.append(rec)
+    return output
+
+def aggFileParents(results):
+    """
+    performs FileParents API aggregation
+
+    :param: results: list JSON records
+    :type: list
+    :return: aggregated list of JSON records
+    """
+    output = []
+    parents = {}
+    for rec in results:
+        lfn = rec['logical_file_name']
+        parent = rec['parent_logical_file_name']
+        if isinstance(parent, list):
+            break
+        parents.setdefault(lfn, []).append(parent)
+    # if we get results from Python server no aggregation is required
```

Comment:
Same comment here. How about not wasting CPU cycles when not needed (thus, when retrieving data from the Python server)?

Reply:
To make it clear, the DBS Client does not know to which server it talks to (if we will have the same URL), I don't want to add hardcoded knowledge which can be only based on URL. As such the code should work with both implementation such that final output of API will be the same. If data comes from Python server it is already aggregated, therefore by looking at first row from results I can break the loop and do nothing. But if data comes from Go server, I need to process all results records to perform aggregation.